Refactored bootstrap components to MUI

OmegaAsh opened the pull request:
Pull-Request for `paretOS`

## Description
- Refactored all of the react-bootstrap code in the ResetPassword file to now use MUI components.

## Relates to
- #169 

## Reviewers
- @mikhael28 
- @jayeclark 


### Screenshots / other info

<img width="887" alt="reset1" src="https://user-images.githubusercontent.com/60603397/163892824-6bc93c4f-8b92-4451-8d8a-3a02ecee9354.png">
<img width="794" alt="reset2" src="https://user-images.githubusercontent.com/60603397/163892827-7058c84d-61c8-47b9-a299-11fa151f6514.png">
<img width="803" alt="reset3" src="https://user-images.githubusercontent.com/60603397/163892828-92bd5b0f-77c2-45f0-b961-a8d6604fd4a2.png">

mikhael28 reviewed (changes requested):
Hey @OmegaAsh this looks great! I do want to request one change - can you please integrate these MUI forms using `react-hook-form` and adding validation to the data entry? Please refer to `Login.js` to see an example of this! 🚀 

We are looking to have all of our forms use `react-hook-form`, because it has better performance - fewer re-renders!

mikhael28 commented:
Tell us about yourself, where are you from and what are your goals in the world of JS?

OmegaAsh commented:
Hey again, thanks for the input, I am currently a 1st year in Comp Sci - I am just getting into open source and web-dev but hopefully I can help! I started to learn React a few weeks ago so I would like to get better at that for now.

I have just updated the ResetPassword file to use React-Hook-Form, and added some basic validation to all of the inputs. I think the login page theme also changed, so I changed ResetPassword to use the theme in there. Did not have too much experience with React hooks but I think it works okay, let me know if there are any more changes you need.

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
     try {
-      await Auth.forgotPasswordSubmit(state.email, state.code, state.password);
-      setState({ ...state, confirmed: true });
+
```
Can you please remove this unnecessary line space? In the next line after the `try` block, there should be some code - shouldn't be any space. My `prettier` extension is complaining.

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+        className={classes.root}
+        onSubmit={handleSubmit(handleSendCodeClick)}
+      >
+        <div>
```
Can you remove this unnecessary div?

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+          />
+        </div>
+        <span className="error">{errors.email && errors.email.message}</span>
+        <div>
```
Can you remove this unnecessary div?

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
-      />
-    </form>
+    <div className="Form">
+      <Container>
```
Can you remove this unnecessary Container element? Or explain why it's needed?

OmegaAsh replied:
Hi again, so when changing between the email entry page and the code confirmation page, when I take the container tag out, the input from the email entry gets carried over to the code confirmation TextField when the code confirmation form is rendered. Not sure why this is, but adding the container tag seems to fix this. If you can suggest another way to fix this issue that would be faster I can do that.

OmegaAsh replied:
I replaced this with a div instead which worked just as well, so the container element doesn't need to be imported

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+          className={classes.root}
+          onSubmit={handleSubmit2(handleConfirmClick)}
+        >
+          <div>
```
Unnecessary div

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+
+          <span className="error">{errors2.code && errors2.code.message}</span>
+          <hr />
+          <div>
```
unnecessary div

OmegaAsh replied:
When I take these divs out, the password and confirm password TextFields join together which maybe is a bit untidy. I think you can achieve the same with the Grid container and Grid item tags, but I am not sure if that would be slower than the divs. Is there another way to do this or would just taking the divs out be okay anyway?
<img width="320" alt="password" src="https://user-images.githubusercontent.com/60603397/164774380-0d6b47b2-fd1a-4a71-8d1f-ccd015563e52.png">

OmegaAsh replied:
I think this is because the break in the code happens after the error message which only gets rendered when an invalid password gets entered.

OmegaAsh replied:
On second look this layout seems fine, so I have removed the div.

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+            {errors2.password && errors2.password.message}
+          </span>
+          <br />
+          <div>
```
unnecessary div

mikhael28 commented on src/profile/ResetPassword.js:
```diff
@@ -1,148 +1,251 @@
+      <Typography variant="h1">{I18n.get("Success!")}</Typography>
+
       <p>Your password has been reset.</p>
       <p>
```
I don't think you need to wrap these Link components inside a <p> tag

mikhael28 reviewed (changes requested):
Great work! I do have some changes to request - please limit the amount of `divs` in the future - you have some divs wrapping TextFields, and you also use a Container in one instance. The more divs we have, the more of a performance hit we will see.

OmegaAsh commented:
Hi again, I removed all the unnecessary divs except for 1, which when removed was causing the input from the previous form to carry over. Other than this I think all the requested changes have been done, let me know if you would like anything else!

mikhael28 reviewed (approved):
Great work! Merging in! We have a special Discord channel for contributors only, if you would like to join - leave your email in a comment!
